server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -283,18 +283,16 @@ private static boolean hasSameMappingHash(FieldCapabilitiesIndexResponse r1, Fie
             && r1.getIndexMappingHash().equals(r2.getIndexMappingHash());
     }
 
-    private FieldCapabilitiesResponse merge(
+    private static FieldCapabilitiesResponse merge(
         Map<String, FieldCapabilitiesIndexResponse> indexResponsesMap,
         CancellableTask task,
         FieldCapabilitiesRequest request,
         List<FieldCapabilitiesFailure> failures
     ) {
         assert ThreadPool.assertCurrentThreadPool(ThreadPool.Names.SEARCH_COORDINATION); // too expensive to run this on a transport worker
         task.ensureNotCancelled();
-        final FieldCapabilitiesIndexResponse[] indexResponses = indexResponsesMap.values()
-            .stream()
-            .sorted(Comparator.comparing(FieldCapabilitiesIndexResponse::getIndexName))
-            .toArray(FieldCapabilitiesIndexResponse[]::new);
+        final FieldCapabilitiesIndexResponse[] indexResponses = indexResponsesMap.values().toArray(new FieldCapabilitiesIndexResponse[0]);
+        Arrays.sort(indexResponses, Comparator.comparing(FieldCapabilitiesIndexResponse::getIndexName));
```

Comment:
Is the idea that this is faster or more GC-friendly that using the sorted stream?  If so it might be worth adding a comment to prevent IDE's pestering future readers to switch back.

Reply:
Yea we save at least one array copy (even for sized stream we copy the result to an `Object[]` array as far as I can see)  + all the objects and indirection from the stream. Mainly did this her because it's less code though (both as text to read + for the compiler) :)